### PR TITLE
fixed issues: refs #409, #447

### PR DESCRIPTION
**Pull request by takei-shg**
## Change

This correction contains to re-create UserIndex table data,
when the screenName or userProfileImageURL is different from the data provided by TwitterService.

Previously, the UserEx data which is stored in the session was not updated,
when the user is already stored in the UserIndex table.
This fix refresh the UserEx in the session, so the screenName & userProfileImageURL are got updated.

I think the occurrence of the change in screenName and Image is rare,
so this correction might not cause the performance problem,
though it recreate the UserIndex data.
(By the way, TwitterLinkage data is updated in every verification.)
## Testing

I didn't create the UnitTest, since Testing VerifyForTwitterAction needs mocking of Cache.get().
Mockito can not mock the static method.
(necessary to use PowerMock??)

I tested by hand by repeating Logout -> Login,
with changing the image-only, screenNamge-only, and image & screenName.

Now, 1 testcase is failed. This seems happened from some previous revisions. Is it right?
[info] TestAdminModifySettingAPI:
[info] - Accessing with admin user **\* FAILED ***
[info]   Expected None, but got Some(value) (TestAdminModifySettingAPI.scala:14)
## Other

I think this correction might not fixes the issue #386.

### CODE REVIEW

**Comment by takei-shg** [comment]
In vagrant-environment, the failed testcase (TestAdminModifySettingAPI) has passed.

**Comment by KengoTODA** [comment]
> [info] TestAdminModifySettingAPI:
> [info] - Accessing with admin user **\* FAILED ***
> [info] Expected None, but got Some(value) (TestAdminModifySettingAPI.scala:14)

This is known problem (#441). To pass this test case, we have to truncate table.
Our Travis CI has already tested this branches, and he said it looks OK.

**Comment by KengoTODA** [comment]
I think this branch has a problem.

Currently this code will override `createdAt` field with current timestamp, and reset `modifiedAt` field with null.
In ideal implementation. But it should be:
1. `createdAt` should be kept.
2. `modifiedAt` should be updated with current timestamp.

Can we improve it? I do not know impact of this problem, but it is better to use field in correct way.
Thanks for your contribution!

**Comment by KengoTODA** [comment]
Here is my suggestion:

``` java
    private UserEx getUserFromTwitterLinkage(PartakeConnection con, IPartakeDAOs daos, UserTwitterLink twitterLinkage) throws DAOException, TwitterException {
        String userId = twitterLinkage.getUserId();
        User user = daos.getUserAccess().find(con, userId);
        if (user == null) {
            user = new User(userId, twitterLinkage.getScreenName(), twitterLinkage.getProfileImageURL(), TimeUtil.getCurrentDateTime(), null);
            daos.getUserAccess().put(con, user);
            user.freeze();
        } else if (!verifyUserProfiles(user, twitterLinkage)) {
            user = new User(user);  // omit if we can
            user.setScreenName(twitterLinkage.getScreenName());
            user.setProfileImageURL(twitterLinkage.getProfileImageURL());
            user.setModifiedAt(TimeUtil.getCurrentDateTime());
            daos.getUserAccess().put(con, user);
            user.freeze();
        }
        return return new UserEx(user, twitterLinkage);;
    }
```

Be careful, I did not compile and test this.

**Comment by KengoTODA** [comment]
Hi @mayah,
do we have to update `modifiedAt` field in this case? I am afraid that this field should be updated only when user modifies some data by own.

If we do not have to update, please notice us. Thanks.

**Comment by takei-shg** [comment]
Thanks @eller86, for pointing out the problem.
Actually, I couldn't understand how the `createdAt` and `modifiedAt` field is maintained and used.

The user would be frozen, so I think `user = new User(user);` is necessary.

**Comment by mayah** [comment]
I think it's OK to leave modifiedAt as is.

**Comment by takei-shg** [comment]
@mayah, @eller86 

Thank you for the instruction.
I modified the codes.
- use ObjectUtils (Here imported from commons-lang 2.6. The app depends on both 2.6 and 3.1.)
- update the user field data mutably.

Tested by hand as the first note.

File: app/in/partake/controller/action/auth/VerifyForTwitterAction.java [hunk 1]
```diff
@@ -119,14 +120,30 @@ private UserTwitterLink updateTwitterLinkage(PartakeConnection con, IPartakeDAOs
...
-        if (user != null)
-            return new UserEx(user, twitterLinkage);
+        if (user == null) {
+            user = new User(userId, twitterLinkage.getScreenName(), twitterLinkage.getProfileImageURL(), TimeUtil.getCurrentDateTime(), null);
+        } else if (!verifyUserProfiles(user, twitterLinkage)) {
+            user = new User(user); 
+            user.setScreenName(twitterLinkage.getScreenName());
+            user.setProfileImageURL(twitterLinkage.getProfileImageURL());
+        }
+        daos.getUserAccess().put(con, user);
```
Comment (KengoTODA) [inline app/in/partake/controller/action/auth/VerifyForTwitterAction.java]: In many case this `put()` is needless. I think we can put this into if block.

**Comment by takei-shg** [comment]
Thanks, @eller86. The code was modified.

**Comment by KengoTODA** [comment]
LGTM :+1: 
We are still able to improve code (`else { if () {` can be `else if () {`) but it is not severe problem.

**Comment by takei-shg** [comment]
@eller86 ,

Thx, I think this code is easy to understand the process, since the check of nullability and verification of data is separated. I agree that It looks a little bit ugly.

**Pull request merged**